geo/src/algorithm/mean_center.rs

```diff
@@ -0,0 +1,694 @@
+use crate::{Centroid, CoordsIter};
+use geo_types::{
+    Geometry, GeometryCollection, Line, LineString, MultiLineString, MultiPoint, MultiPolygon,
+    Point, Polygon, Rect, Triangle,
+};
+
+/// Calculate the (Weighted) Mean Center of a Geometry
+///
+/// The mean center of a geometry is a measure of central tendancy
+/// of a set of geometry. It is calculated by taking the average
+/// of all x and y values in the geometry. For [`Point`], [`Line`], [`LineString`],
+/// [`Polygon`], [`Triangle`], and [`Rect`] this is a simple alias to [`Centroid`].
+/// However, [`MultiLineString`], [`MultiPolygon`], and [`GeometryCollection`],
+/// the mean center is found by taking the centroid of each geometry and
+/// calculating the mean center of the centroids.
+///
+/// The weighted mean center applies a weight to each geometry. The weight is then
+/// applied to the value of the coordinate used used in the calculation of the
+/// mean center.
+///
+/// For [`Line`], [`Triangle`], [`Rect`], [`LineString`], [`Polygon`], and [`MultiPoint`], weights
+/// are applied  to each coordinate in the geometry. For [`MultiLineString`], [`MultiPolygon`],
+/// and [`GeometryCollection`], the weights are applied to the centroid of each component
+/// geometry. If a geometry does not have a centroid, it is filtered out.
+///
+/// The `weights` argument takes a slice which is used as an iterator which is cycled.
+/// If there are fewer weights than coordinates, the weights will be recycled until the
+/// calculation has completed.
```

Comment:
> If there are fewer weights than coordinates, the weights will be recycled until the calculation has completed.

Is there a useful reason for this cycle behavior or some precedent? 

Or is this more like "do something, anything, so long as we don't explode"?